### PR TITLE
Add `@Init(assignee:type:)` to (awkwardly) support property wrappers

### LINKED ISSUE #11: Add `@Init(assignee:type:)` to (awkwardly) support property wrappers
MemberwiseInit can be made to support property wrappers like SwiftUI's `@Binding` with the addition of two new arguments:

```swift
import SwiftUI

@MemberwiseInit
struct CounterView: View {
  @Init(assignee: "self._count", type: Binding<Int>)
  @Binding var count: Int
}
```

With that, MemberwiseInit can provide the expected initializer for a `@Binding`-wrapped property:

```swift
internal init(
  property: Binding<Int>
) {
  self._count = count
}
```

This expressiveness may also support the use of other framework-vended property wrappers, custom property wrappers and macros.

### PR DESCRIPTION
**Pull request by gohanlon**
Resolves #11.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,6 +146,20 @@ Attach to member property declarations of a struct, actor, or class that `@Membe
...
+  <br> Override the target property to which the initializer argument should be assigned. By default, a property named “property” has the assignee `self.property`, as demonstrated in `self.property = property`.
+
+* `@Init(type: Any.Type)`
+  <br> Override the type of the argument in the initializer.
+
+* `@Init(assignee: String, type: Any.Type)`
+  <br> Combine `assignee` and `type` to support usage with property wrappers:
+
+  ```swift
+  @Init(assignee: "self._isOn", type: Binding<Bool>)
```
Comment (davdroman) [inline README.md:159]: Can the macro not read the property name and assume it's always `self._propertyName`? Are there caveats to this? 🤔

Comment (davdroman) [reply README.md:159]: Oh I see that `type` is more of a general purpose thing than anything specifically related to property wrappers. Interesting.

I wonder if we can have a specific property wrapper argument somehow. Like:

```swift
@Init(propertyWrapperType: Binding<Bool>)
```

Comment (gohanlon) [reply README.md:159]: Usually, I think that would work. But, MemberwiseInit would technically have to assume that `@Binding` is indeed a `SwiftUI.@Binding` property wrapper and not some other attribute/macro. I've considered adding a `mode` that could make that assumption reasonable:

```swift
import SwiftUI

@MemberwiseInit(.public, mode: .swiftUI)
struct CounterView: View {
  @Binding public var count = 0

  var body: some View { … }
}
```

I haven't completely thought through the implications there. My primary concern is how to support multiple versions of Swift and SwiftUI over time.

Thanks, and definitely do let me know your thoughts!

Comment (gohanlon) [reply README.md:159]: (Our latest comments crossed.) That's an interesting idea! I'll think some on that tonight. I definitely agree that providing the assignee as a string is the most awkward part of the current solution.

Comment (gohanlon) [reply README.md:159]: I started #14 with some draft thoughts on how to better support property-wrapped members.

**Pull request merged**